definitions/features/foreman_tasks.rb

```diff
@@ -19,4 +21,58 @@ def paused_tasks_count
     # SQL
     5
   end
+
+  def old_tasks_count
+    feature(:foreman_database).query(<<-SQL).first['count'].to_i
+     SELECT count(*) AS count FROM foreman_tasks_tasks WHERE #{old_tasks_condition}
+    SQL
+  end
+
+  def delete_old_tasks
+    feature(:foreman_database).psql(<<-SQL)
+     BEGIN;
+       DELETE FROM dynflow_steps USING foreman_tasks_tasks WHERE (foreman_tasks_tasks.external_id = dynflow_steps.execution_plan_uuid) AND #{old_tasks_condition};
+       DELETE FROM dynflow_actions USING foreman_tasks_tasks WHERE (foreman_tasks_tasks.external_id = dynflow_actions.execution_plan_uuid) AND #{old_tasks_condition};
+       DELETE FROM dynflow_execution_plans USING foreman_tasks_tasks WHERE (foreman_tasks_tasks.external_id = dynflow_execution_plans.uuid) AND #{old_tasks_condition};
+       DELETE FROM foreman_tasks_tasks WHERE #{old_tasks_condition};
+     COMMIT;
+    SQL
+
+    old_tasks_count
+  end
+
+  def backup_tasks
+    backup_table('dynflow_execution_plans', 'uuid') { || yield(status) }
+    backup_table('dynflow_steps') { |status| yield(status) }
+    backup_table('dynflow_actions') { |status| yield(status) }
+
+    yield('Backup Tasks [running]')
+    export_csv("SELECT * FROM foreman_tasks_tasks WHERE #{old_paused_tasks_condition}",
+               '$DIR/foreman_tasks_tasks.csv')
```

Comment:
It actually goes as a `file` argument to `export_csv` which then executes the string. So $DIR will be environment variable if any is set. Else the files will be saved to /
Which is not a good idea. What would be the best place to write the backup? One I can think of is user home(~/<some-folder>) or `/var/local/`